### PR TITLE
Better CI integration of a docker build

### PR DESCRIPTION
**Pull request by xnull**
## Overview

Description:
- Docker build process has been extracted from maven and now is in .ci/infrastructure-docker-build.sh which makes build process more flexible simple and straightforward
- protobuf plugin fix
- migrate to a newer version of maven-compiler-plugin  (the old version didn't print the reason of compilation error in case of lombok annotation-processing failures)
- Lombok needed to be updated because it caused compilation error in the result of fix in the protobuf plugin  

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/pom.xml [hunk 1]
```diff
@@ -380,31 +92,6 @@
                     <shadedClassifierName>shaded</shadedClassifierName>
                 </configuration>
             </plugin>
-            <plugin>
-                <groupId>com.github.os72</groupId>
-                <artifactId>protoc-jar-maven-plugin</artifactId>
```
Comment (hisundar) [inline infrastructure/pom.xml:385]: Does this mean protoc must be installed already in any environment that needs to build corfu? What if a build environment does not have protoc compiler present?

Comment (xnull) [reply infrastructure/pom.xml:385]: nope, you can see that I have deleted a second copy of the plugin. The first one is still present in the pom.
So I just deleted an extra /duplicate plugin

Comment (hisundar) [reply infrastructure/pom.xml:385]: thanks

**Review by hisundar — APPROVED** [review]
LGTM

**Review by Lujie1996 — APPROVED** [review]

**Review by pankti-m — APPROVED** [review]

**Pull request merged**